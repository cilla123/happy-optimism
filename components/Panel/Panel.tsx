import React, { PureComponent } from 'react';
import classnames from 'classnames';

export interface PanelProps {
  prefixCls?: string;
  className?: string;
}

export default class Panel extends PureComponent<PanelProps, {}> {

  static Header: any;
  static Body: any;
  static Footer: any;

  static defaultProps = {
    prefixCls: 'za-panel',
  };

  render() {
    const { prefixCls, className, children } = this.props;
    const cls = classnames(`${prefixCls}`, className);

    return <div className={cls}>{children}</div>;
  }
}
