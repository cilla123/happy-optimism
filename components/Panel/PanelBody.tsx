import React, { PureComponent } from 'react';
import classnames from 'classnames';

export interface PanelBodyProps {
  prefixCls?: string;
  className?: string;
}

export default class PanelBody extends PureComponent<PanelBodyProps, {}> {

  static defaultProps = {
    prefixCls: 'za-panel',
  };

  render() {
    const { prefixCls, className, children, ...others } = this.props;
    const cls = classnames(`${prefixCls}-body`, className);

    return <div {...others} className={cls}>{children}</div>;
  }
}
