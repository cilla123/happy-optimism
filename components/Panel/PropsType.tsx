export interface BasePanelHeaderProps {
  title?: any;
  more?: any;
}

export interface BasePanelFooterProps {
  title?: any;
  more?: any;
}
