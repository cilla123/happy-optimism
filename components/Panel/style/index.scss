@import "../../style/core/index";
@import "../../style/themes/default";
@import "./component";
