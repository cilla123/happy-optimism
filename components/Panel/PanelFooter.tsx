import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BasePanelFooterProps } from './PropsType';

export interface PanelFooterProps extends BasePanelFooterProps {
  prefixCls?: string;
  className?: string;
}

export default class PanelFooter extends PureComponent<PanelFooterProps, {}> {

  static defaultProps = {
    prefixCls: 'za-panel',
  };

  render() {
    const { prefixCls, className, title, more } = this.props;
    const cls = classnames(`${prefixCls}-footer`, className);

    return (
      <div className={cls}>
        {title && <div className={`${prefixCls}-title`}>{title}</div>}
        {more && <div className={`${prefixCls}-more`}>{more}</div>}
      </div>
    );
  }
}
