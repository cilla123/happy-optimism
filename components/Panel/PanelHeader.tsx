import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BasePanelHeaderProps } from './PropsType';

export interface PanelHeaderProps extends BasePanelHeaderProps {
  prefixCls?: string;
  className?: string;
}

export default class PanelHeader extends PureComponent<PanelHeaderProps, {}> {

  static defaultProps = {
    prefixCls: 'za-panel',
  };

  render() {
    const { prefixCls, className, title, more } = this.props;
    const cls = classnames(`${prefixCls}-header`, className);

    return (
      <div className={cls}>
        {title && <div className={`${prefixCls}-title`}>{title}</div>}
        {more && <div className={`${prefixCls}-more`}>{more}</div>}
      </div>
    );
  }
}
