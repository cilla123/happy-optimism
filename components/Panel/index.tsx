import Panel from './Panel';
import PanelHeader from './PanelHeader';
import PanelBody from './PanelBody';
import PanelFooter from './PanelFooter';

Panel.Header = PanelHeader;
Panel.Body = PanelBody;
Panel.Footer = PanelFooter;

export default Panel;
