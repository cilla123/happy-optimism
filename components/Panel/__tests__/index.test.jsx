import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Panel from '../index';

describe('Panel', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Panel>
        <Panel.Header title="title" more="more" />
        <Panel.Body>body</Panel.Body>
        <Panel.Footer title="title" more="more" />
      </Panel>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
