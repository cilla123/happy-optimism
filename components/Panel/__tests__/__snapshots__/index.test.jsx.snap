// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly 1`] = `
<div
  class="za-panel"
>
  <div
    class="za-panel-header"
  >
    <div
      class="za-panel-title"
    >
      title
    </div>
    <div
      class="za-panel-more"
    >
      more
    </div>
  </div>
  <div
    class="za-panel-body"
  >
    body
  </div>
  <div
    class="za-panel-footer"
  >
    <div
      class="za-panel-title"
    >
      title
    </div>
    <div
      class="za-panel-more"
    >
      more
    </div>
  </div>
</div>
`;
